ui/DesignSystem/PasswordInput.svelte

```diff
@@ -30,6 +30,12 @@
     inputElement && inputElement.focus();
   };
 
+  export const setType = (type: string): void => {
+    if (inputElement) {
+      inputElement.type = type;
+    }
+  };
+
```

Comment:
I'd prefer if we didn't expose this component's internals.
How about doing it via a prop?

```diff
diff --git a/ui/DesignSystem/PasswordInput.svelte b/ui/DesignSystem/PasswordInput.svelte
index da422d54..13143ccd 100644
--- a/ui/DesignSystem/PasswordInput.svelte
+++ b/ui/DesignSystem/PasswordInput.svelte
@@ -25,17 +25,12 @@
   export let hint = "";
   export let spellcheck: boolean = false;
   export let autofocus: boolean = false;
+  export let visible: boolean = false;

   export const focus = (): void => {
     inputElement && inputElement.focus();
   };

-  export const setType = (type: string): void => {
-    if (inputElement) {
-      inputElement.type = type;
-    }
-  };
-
   let inputElement: HTMLInputElement | undefined = undefined;

   const dispatch = createEventDispatcher();
@@ -56,6 +51,9 @@
   };

   $: showHint = hint.length > 0 && value.length === 0;
+  $: if(inputElement) {
+    inputElement.type = visible ? "text" : "password"
+  }
 </script>

 <style>
diff --git a/ui/Screen/Lock.svelte b/ui/Screen/Lock.svelte
index 90384424..ecefcfff 100644
--- a/ui/Screen/Lock.svelte
+++ b/ui/Screen/Lock.svelte
@@ -61,13 +61,11 @@
   const resetCheck = () => {
     if (passphrase.length === 0) {
       visible = false;
-      input.setType("password");
     }
   };

   const togglePassphraseDisplay = () => {
     visible = !visible;
-    input.setType(visible ? "text" : "password");
   };
 </script>

@@ -107,6 +105,7 @@
     <PasswordInput
       bind:this={input}
       autofocus
+      {visible}
       placeholder="Enter your passphrase"
       bind:value={passphrase}
       disabled={unlockInProgress}
diff --git a/ui/Screen/Onboarding/EnterPassphrase.svelte b/ui/Screen/Onboarding/EnterPassphrase.svelte
index fdb15733..dbf6438d 100644
--- a/ui/Screen/Onboarding/EnterPassphrase.svelte
+++ b/ui/Screen/Onboarding/EnterPassphrase.svelte
@@ -112,8 +112,6 @@

   const togglePassphraseDisplay = () => {
     visible = !visible;
-    passphraseInput.setType(visible ? "text" : "password");
-    repeatedPassphraseInput.setType(visible ? "text" : "password");
   };

   const onKeydown = (event: KeyboardEvent) => {
@@ -135,8 +133,6 @@
   const resetCheck = () => {
     if (passphrase.length === 0) {
       visible = false;
-      passphraseInput.setType("password");
-      repeatedPassphraseInput.setType("password");
     }
   };
 </script>
@@ -191,6 +187,7 @@
     </p>

     <PasswordInput
+      {visible}
       bind:this={passphraseInput}
       on:enter={() => {
         repeatedPassphraseInput.focus();
@@ -207,6 +204,7 @@
     <div class="repeat" hidden={!passphrase}>
       <p style="margin-bottom: 0.5rem;">And enter it again, just to be safe.</p>
       <PasswordInput
+        {visible}
         bind:this={repeatedPassphraseInput}
         on:enter={next}
         on:change={resetCheck}
```

Reply:
Thanks @rudolfs , I have made the necessary changed - tested and verified.